### PR TITLE
Parse request implemented

### PR DESCRIPTION
**Pull request by nvandoorn**
Parse request is implemented to spec to the best of my knowledge. I also added some helper methods for debugging.

### CODE REVIEW

File: app/parse_request.js [hunk 1]
```diff
@@ -56,29 +64,42 @@ exports.parseRequest = reqBody => {
...
-    const toReturn = [];
-    for(let i = 0; i < width; i++){
-        const row = []
-        for(let j = 0; j < height; j++){
-            row.push('hi')
+    const occupied = getAllOccupiedSpaces(snakes);
+    const matrix = [];
+    console.log('Occupied spaces on game board:');
+    console.log(occupied);
+    for(let j = 0; j < height; j++){
```
Comment (nvandoorn) [inline app/parse_request.js]: I'm not super stoked that this uses a nested for loop, was hoping for a cleaner functional solution but eh good enough for a hackathon right?

File: app/parse_request.js [hunk 2]
```diff
@@ -56,29 +64,42 @@ exports.parseRequest = reqBody => {
...
-const prettyPrintMatrix = matrix => matrix.map(array => console.log(stringifyArray(array)));
-
-const matrix = generateMatrix(null, 20, 20);
-prettyPrintMatrix(matrix);
+// TODO I got really close to a map/functional
+// solution that was way cleaner, abandoned it for now
+const getAllOccupiedSpaces = snakes => {
+    const coords = [];
+    for(let snake of snakes){
+        for(let coord of snake.coords){
```
Comment (nvandoorn) [inline app/parse_request.js]: Same here.

File: app/parse_request.js [hunk 3]
```diff
@@ -29,20 +29,77 @@
...
+    const ourSnake = getOurSnake(snakes, you).coords;
+    return{
+        head: ourSnake[0],
+        tail: ourSnake.slice(-1)[0] // TODO verify that the last coord is tail
+    }
+}
+
+const pointEquals = (point, x, y) => point[0] === x && point[1] === y;
+
+const generateMatrix = (snakes, width, height) => {
```
Comment (MartinWan) [inline app/parse_request.js]: style opinion: just make a function

File: app/parse_request.js [hunk 4]
```diff
@@ -29,20 +29,77 @@
...
+    const height = reqBody.height;
+    const you = reqBody.you;
+    const snakes = reqBody.snakes;
+    const toReturn = getOurHeadTail(snakes, you);
+    toReturn.food = reqBody.food;
+    toReturn.matrix = generateMatrix(snakes, width, height);
+    return toReturn;
 };
+
+const getOurSnake = (snakes, you) => snakes.filter(snake => snake.id === you)[0];
```
Comment (MartinWan) [inline app/parse_request.js]: .find()

File: app/parse_request.js (hunk 3 above)
Comment (nvandoorn) [inline app/parse_request.js]: Use `function myFunc()` instead?

File: app/parse_request.js (hunk 2 above)
Comment (MartinWan) [inline app/parse_request.js]: .filter().length > 0 can be released with .some(predicate function)

File: app/parse_request.js (hunk 3 above)
Comment (MartinWan) [inline app/parse_request.js]: Yeah, I think that's more standard practice. Usually arrow functions are for anonomous functions

Comment (nvandoorn) [inline app/parse_request.js]: Sure. I'm going to keep the arrow functions for the one liners (no `return` that way)

File: app/parse_request.js (hunk 2 above)
Comment (nvandoorn) [inline app/parse_request.js]: Not sure that I understand this, please clarify.

**Pull request merged**